ipalib/install/certmonger.py

```diff
@@ -317,23 +317,29 @@ def request_and_wait_for_cert(
 
 
 def request_cert(
-        nssdb, nickname, subject, principal, passwd_fname=None,
-        dns=None, ca='IPA', profile=None, pre_command=None, post_command=None):
+        certpath, nickname, subject, principal, passwd_fname=None,
+        dns=None, ca='IPA', profile=None,
+        pre_command=None, post_command=None, storage='NSSDB'):
     """
     Execute certmonger to request a server certificate.
 
     ``dns``
         A sequence of DNS names to appear in SAN request extension.
     """
+    if storage == 'FILE':
+        certfile, keyfile = certpath
```

Comment:
The API for file based certmonger requests is counter intuitive. I'd rather have an additional argument ```keyfile=None``` with a check ```if storage == 'FILE' and keyfile is None: raise ValueError('keyfile is required for file storage')```